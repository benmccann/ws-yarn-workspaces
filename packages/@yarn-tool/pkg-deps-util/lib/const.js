"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumInstallTypesErrorCode = void 0;
var EnumInstallTypesErrorCode;
(function (EnumInstallTypesErrorCode) {
    EnumInstallTypesErrorCode[EnumInstallTypesErrorCode["SUCCESS"] = 0] = "SUCCESS";
    EnumInstallTypesErrorCode[EnumInstallTypesErrorCode["NOT_EXISTS"] = 1] = "NOT_EXISTS";
    EnumInstallTypesErrorCode[EnumInstallTypesErrorCode["DEPRECATED"] = 2] = "DEPRECATED";
    EnumInstallTypesErrorCode[EnumInstallTypesErrorCode["SKIP"] = 3] = "SKIP";
})(EnumInstallTypesErrorCode || (exports.EnumInstallTypesErrorCode = EnumInstallTypesErrorCode = {}));
//# sourceMappingURL=const.js.map