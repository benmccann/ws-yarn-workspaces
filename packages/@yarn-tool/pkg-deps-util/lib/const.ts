export const enum EnumInstallTypesErrorCode
{
	NOT_EXISTS = 1,
	DEPRECATED = 2,

	SKIP = 3,
}
