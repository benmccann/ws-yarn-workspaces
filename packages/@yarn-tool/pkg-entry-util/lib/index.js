"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.pkgExportsVerify = exports.pkgExportsAddPJsonEntry = void 0;
var exports_1 = require("./field/exports");
Object.defineProperty(exports, "pkgExportsAddPJsonEntry", { enumerable: true, get: function () { return exports_1.pkgExportsAddPJsonEntry; } });
Object.defineProperty(exports, "pkgExportsVerify", { enumerable: true, get: function () { return exports_1.pkgExportsVerify; } });
//# sourceMappingURL=index.js.map