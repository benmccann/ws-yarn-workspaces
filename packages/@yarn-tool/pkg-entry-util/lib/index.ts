export { fixPkgBinField } from './field/bin';

export { fixPublishConfig } from './field/publishConfig';

export { pkgExportsAddPJsonEntry, pkgExportsVerify } from './field/exports';
