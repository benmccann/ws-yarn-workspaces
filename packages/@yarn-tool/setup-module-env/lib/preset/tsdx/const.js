"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumTsdx = void 0;
var EnumTsdx;
(function (EnumTsdx) {
    EnumTsdx["keyword"] = "create-by-tsdx";
})(EnumTsdx = exports.EnumTsdx || (exports.EnumTsdx = {}));
//# sourceMappingURL=const.js.map