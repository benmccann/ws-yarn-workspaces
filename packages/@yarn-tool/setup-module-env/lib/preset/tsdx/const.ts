
export const enum EnumTsdx
{
	keyword = 'create-by-tsdx'
}
