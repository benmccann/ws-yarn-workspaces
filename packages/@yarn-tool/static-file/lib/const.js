"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultCopyStaticFilesWsRootOnly = exports.defaultCopyStaticFilesRootOnly = exports.defaultCopyStaticFiles = void 0;
const _defaultCopyStaticFiles = [
    ['.npmignore', 'file/npmignore'],
    ['.gitignore', 'file/gitignore'],
    ['.eslintignore', 'file/eslintignore'],
    ['.nvmrc', 'file/nvmrc'],
    ['.browserslistrc', 'file/browserslistrc'],
    ['tsconfig.json.tpl', 'file/tsconfig.json.tpl', 'tsconfig.json'],
    ['test/tsconfig.json.tpl', 'file/test/tsconfig.json.tpl', 'test/tsconfig.json'],
    ['tsconfig.esm.json.tpl', 'file/tsconfig.esm.json.tpl', 'tsconfig.esm.json'],
    ['tsconfig.tsdx.json.tpl', 'file/tsconfig.tsdx.json.tpl', 'tsconfig.tsdx.json'],
    ['.eslintrc.json.tpl', 'file/eslintrc.json.tpl', '.eslintrc.json'],
    ['README.md', 'file/README.md'],
    ['.nycrc', 'file/nycrc'],
    ['.mocharc.yml', 'file/mocharc.yml'],
    ['jest.config.js', 'file/jest.config.js'],
    ['.nowignore', 'file/nowignore'],
    ['now.json.tpl', 'file/now.json.tpl', 'now.json'],
    ['.npmrc.tpl', 'file/npmrc', '.npmrc'],
    ['tsdx.config.js.tpl', 'file/tsdx.config.js', 'tsdx.config.js'],
    ['tsc-multi.json.tpl', 'file/tsc-multi.json.tpl', 'tsc-multi.json'],
    ['test/__root.ts', 'file/test/__root.ts'],
    ['.editorconfig.tpl', 'file/tpl.editorconfig'],
    //['changelog-option.js.tpl', 'file/changelog-option.js', 'changelog-option.js'],
];
const _defaultCopyStaticFilesRootOnly = [
    ['lerna.json.tpl', 'file/lerna.json.tpl', 'lerna.json'],
    ['pnpm-workspace.yaml.tpl', 'file/pnpm-workspace.yaml', 'pnpm-workspace.yaml'],
    ['.github/workflows/coverage.yml', 'file/github/workflows/coverage.yml'],
    ['.github/workflows/action-yarnlock-dedupe.yml', 'file/github/workflows/action-yarnlock-dedupe.yml'],
    ['.github/workflows/build.yml', 'file/github/workflows/build.yml'],
    ['tsconfig.json', 'file/tsconfig.json.tpl', 'tsconfig.json'],
    ['.eslintrc.json', 'file/eslintrc.json.tpl', '.eslintrc.json'],
    //['changelog-option.js', 'file/changelog-option.js', 'changelog-option.js.tpl'],
    ['.yarnrc.yml.tpl', 'file/root/yarnrc.yml', '.yarnrc.yml'],
];
const _defaultCopyStaticFilesWsRootOnly = [
    ['lerna.json.tpl', 'file/lerna.json.tpl'],
    ['pnpm-workspace.yaml', 'file/pnpm-workspace.yaml'],
    ['tsconfig.json', 'file/tsconfig.json.tpl'],
    ['tsc-multi.json.tpl', 'file/tsc-multi.json.tpl', 'tsc-multi.json'],
    ['__root_ws.ts', 'file/ws-root/__root_ws.ts'],
];
exports.defaultCopyStaticFiles = Object.freeze(_defaultCopyStaticFiles);
exports.defaultCopyStaticFilesRootOnly = Object.freeze(_defaultCopyStaticFilesRootOnly);
exports.defaultCopyStaticFilesWsRootOnly = Object.freeze(_defaultCopyStaticFilesWsRootOnly);
//# sourceMappingURL=const.js.map