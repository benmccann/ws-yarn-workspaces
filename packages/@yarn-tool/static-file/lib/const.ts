import { _Key, IStaticFilesKey, IStaticFilesMapArray } from './types';

const _defaultCopyStaticFiles = [

	['.npmignore', 'file/npmignore'],
	['.gitignore', 'file/gitignore'],

	['.eslintignore', 'file/eslintignore'],

	['.nvmrc', 'file/nvmrc'],
	['.browserslistrc', 'file/browserslistrc'],

	['tsconfig.json.tpl', 'file/tsconfig.json.tpl', 'tsconfig.json'],

	['test/tsconfig.json.tpl', 'file/test/tsconfig.json.tpl', 'test/tsconfig.json'],

	['tsconfig.esm.json.tpl', 'file/tsconfig.esm.json.tpl', 'tsconfig.esm.json'],

	['tsconfig.tsdx.json.tpl', 'file/tsconfig.tsdx.json.tpl', 'tsconfig.tsdx.json'],

	['.eslintrc.json.tpl', 'file/eslintrc.json.tpl', '.eslintrc.json'],

	['README.md', 'file/README.md'],

	['.nycrc', 'file/nycrc'],

	['.mocharc.yml', 'file/mocharc.yml'],
	['jest.config.js', 'file/jest.config.js'],

	['.nowignore', 'file/nowignore'],

	['now.json.tpl', 'file/now.json.tpl', 'now.json'],

	['.npmrc.tpl', 'file/npmrc', '.npmrc'],

	['tsdx.config.js.tpl', 'file/tsdx.config.js', 'tsdx.config.js'],

] as const;

const _defaultCopyStaticFilesRootOnly = [

	['lerna.json.tpl', 'file/lerna.json.tpl', 'lerna.json'],

	['pnpm-workspace.yaml.tpl', 'file/pnpm-workspace.yaml', 'pnpm-workspace.yaml'],

	['.github/workflows/coverage.yml', 'file/github/workflows/coverage.yml'],

	['.github/workflows/action-yarnlock-dedupe.yml', 'file/github/workflows/action-yarnlock-dedupe.yml'],

	['tsconfig.json', 'file/tsconfig.json.tpl', 'tsconfig.json'],

] as const;

export const defaultCopyStaticFiles = Object.freeze(_defaultCopyStaticFiles) as any as IStaticFilesMapArray<_Key<typeof _defaultCopyStaticFiles>>;

export const defaultCopyStaticFilesRootOnly = Object.freeze(_defaultCopyStaticFilesRootOnly) as any as IStaticFilesMapArray<_Key<typeof _defaultCopyStaticFilesRootOnly>>;
