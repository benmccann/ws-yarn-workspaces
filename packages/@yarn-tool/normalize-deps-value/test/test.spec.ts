//@noUnusedParameters:false

import { _getNpaResult, normalizeDepsValue, normalizeResultToDepsValue } from '../index';
import { EnumSemverVersion } from '@lazy-node/semver-ampersand/lib/const';

beforeAll(async () =>
{

});

describe(`normalizeDepsValue`, () =>
{
	[
		'1.0 && (2.0    || 3.1)',
		'1.0 (2.0 || 3.1)',
		'>x 2.x || * || <x || ~1.2.3beta ^4.0.0-dev.20200615 && 4.0.0-dev.20200615 < 4.0.0-dev.20200800 || 4.0.x - 4.0.2 || (4.0.x - 4.0.2) || >= 1 || =1 || ~> 2.0.0 || 1.2.3 - 1.2.4 - 1.2.5 || >=1.2.3 <=1.2.4',
		'gitlab:user/foo-js',
		'gitlab:user/foo-js#bar/baz',
		'gitlab:user..blerg--/..foo-js# . . . . . some . tags / / /',
		'gitlab:user/foo-js#bar/baz/bin',
		'foo@gitlab:user/foo-js',
		'git+ssh://git@gitlab.com/user/foo#1.2.3',
		'https://gitlab.com/user/foo.git',
		'@foo/bar@git+ssh://gitlab.com/user/foo',
		'*',
		'',
		'x',
		'1',
		'0',
		' ',
		'.',
	].forEach(input =>
	{
		test(input === EnumSemverVersion.ANY ? 'empty' : input === ' ' ? 'space' : input, () =>
		{

			let result = _getNpaResult(input);
			let actual = normalizeResultToDepsValue(result);

			expect(result).toMatchSnapshot();
			expect(actual).toMatchSnapshot();

		});
	})

})
