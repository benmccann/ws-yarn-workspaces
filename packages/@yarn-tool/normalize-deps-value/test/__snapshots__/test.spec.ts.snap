// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeDepsValue * 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "*",
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@*",
  "rawSpec": "*",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "type": "range",
  "where": undefined,
}
`;

exports[`normalizeDepsValue * 2`] = `"*"`;

exports[`normalizeDepsValue . 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "G:\\Users\\The Project\\nodejs-yarn\\ws-yarn-workspaces2\\packages\\@yarn-tool\\normalize-deps-value",
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@.",
  "rawSpec": ".",
  "registry": undefined,
  "saveSpec": "file:",
  "scope": "@fake",
  "type": "directory",
  "where": "G:\\Users\\The Project\\nodejs-yarn\\ws-yarn-workspaces2\\packages\\@yarn-tool\\normalize-deps-value",
}
`;

exports[`normalizeDepsValue . 2`] = `"."`;

exports[`normalizeDepsValue >x 2.x || * || <x || ~1.2.3beta ^4.0.0-dev.20200615 && 4.0.0-dev.20200615 < 4.0.0-dev.20200800 || 4.0.x - 4.0.2 || (4.0.x - 4.0.2) || >= 1 || =1 || ~> 2.0.0 || 1.2.3 - 1.2.4 - 1.2.5 || >=1.2.3 <=1.2.4 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": ">x 2.x || * || <x || ~1.2.3beta ^4.0.0-dev.20200615 && 4.0.0-dev.20200615 < 4.0.0-dev.20200800 || 4.0.x - 4.0.2 || (4.0.x - 4.0.2) || >= 1 || =1 || ~> 2.0.0 || 1.2.3 - 1.2.4 - 1.2.5 || >=1.2.3 <=1.2.4",
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@>x 2.x || * || <x || ~1.2.3beta ^4.0.0-dev.20200615 && 4.0.0-dev.20200615 < 4.0.0-dev.20200800 || 4.0.x - 4.0.2 || (4.0.x - 4.0.2) || >= 1 || =1 || ~> 2.0.0 || 1.2.3 - 1.2.4 - 1.2.5 || >=1.2.3 <=1.2.4",
  "rawSpec": ">x 2.x || * || <x || ~1.2.3beta ^4.0.0-dev.20200615 && 4.0.0-dev.20200615 < 4.0.0-dev.20200800 || 4.0.x - 4.0.2 || (4.0.x - 4.0.2) || >= 1 || =1 || ~> 2.0.0 || 1.2.3 - 1.2.4 - 1.2.5 || >=1.2.3 <=1.2.4",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "type": "range",
  "where": undefined,
}
`;

exports[`normalizeDepsValue >x 2.x || * || <x || ~1.2.3beta ^4.0.0-dev.20200615 && 4.0.0-dev.20200615 < 4.0.0-dev.20200800 || 4.0.x - 4.0.2 || (4.0.x - 4.0.2) || >= 1 || =1 || ~> 2.0.0 || 1.2.3 - 1.2.4 - 1.2.5 || >=1.2.3 <=1.2.4 2`] = `">x 2.x || * || <x || ~1.2.3beta ^4.0.0-dev.20200615 4.0.0-dev.20200615 <4.0.0-dev.20200800 || >=4.0.0 <=4.0.2 || (4.0.x - 4.0.2) || >=1 || =1 || ~>2.0.0 || 1.2.3 - 1.2.4 - 1.2.5 || >=1.2.3 <=1.2.4"`;

exports[`normalizeDepsValue @foo/bar@git+ssh://gitlab.com/user/foo 1`] = `
{
  "escapedName": "@foo%2fbar",
  "fetchSpec": "ssh://git@gitlab.com/user/foo.git",
  "gitCommittish": null,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@foo/bar",
  "raw": "@foo/bar@git+ssh://gitlab.com/user/foo",
  "rawSpec": "git+ssh://gitlab.com/user/foo",
  "registry": undefined,
  "saveSpec": "git+ssh://git@gitlab.com/user/foo.git",
  "scope": "@foo",
  "type": "git",
  "where": undefined,
}
`;

exports[`normalizeDepsValue @foo/bar@git+ssh://gitlab.com/user/foo 2`] = `"git+ssh://gitlab.com/user/foo"`;

exports[`normalizeDepsValue 0 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "0",
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@0",
  "rawSpec": "0",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "type": "range",
  "where": undefined,
}
`;

exports[`normalizeDepsValue 0 2`] = `"0"`;

exports[`normalizeDepsValue 1 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "1",
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@1",
  "rawSpec": "1",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "type": "range",
  "where": undefined,
}
`;

exports[`normalizeDepsValue 1 2`] = `"1"`;

exports[`normalizeDepsValue 1.0 && (2.0    || 3.1) 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "1.0 && (2.0    || 3.1)",
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@1.0 && (2.0    || 3.1)",
  "rawSpec": "1.0 && (2.0    || 3.1)",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "type": "range",
  "where": undefined,
}
`;

exports[`normalizeDepsValue 1.0 && (2.0    || 3.1) 2`] = `"1.0 (2.0 || 3.1)"`;

exports[`normalizeDepsValue 1.0 (2.0 || 3.1) 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "1.0 (2.0 || 3.1)",
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@1.0 (2.0 || 3.1)",
  "rawSpec": "1.0 (2.0 || 3.1)",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "type": "range",
  "where": undefined,
}
`;

exports[`normalizeDepsValue 1.0 (2.0 || 3.1) 2`] = `"1.0 (2.0 || 3.1)"`;

exports[`normalizeDepsValue empty 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "*",
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@",
  "rawSpec": "*",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "type": "range",
  "where": undefined,
}
`;

exports[`normalizeDepsValue empty 2`] = `"*"`;

exports[`normalizeDepsValue foo@gitlab:user/foo-js 1`] = `
{
  "escapedName": "foo",
  "fetchSpec": null,
  "gitCommittish": null,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "foo",
  "raw": "foo@gitlab:user/foo-js",
  "rawSpec": "gitlab:user/foo-js",
  "registry": undefined,
  "saveSpec": "gitlab:user/foo-js",
  "scope": undefined,
  "type": "git",
  "where": undefined,
}
`;

exports[`normalizeDepsValue foo@gitlab:user/foo-js 2`] = `"gitlab:user/foo-js"`;

exports[`normalizeDepsValue git+ssh://git@gitlab.com/user/foo#1.2.3 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "ssh://git@gitlab.com/user/foo.git",
  "gitCommittish": "1.2.3",
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@git+ssh://git@gitlab.com/user/foo#1.2.3",
  "rawSpec": "git+ssh://git@gitlab.com/user/foo#1.2.3",
  "registry": undefined,
  "saveSpec": "git+ssh://git@gitlab.com/user/foo.git#1.2.3",
  "scope": "@fake",
  "type": "git",
  "where": undefined,
}
`;

exports[`normalizeDepsValue git+ssh://git@gitlab.com/user/foo#1.2.3 2`] = `"git+ssh://git@gitlab.com/user/foo#1.2.3"`;

exports[`normalizeDepsValue gitlab:user..blerg--/..foo-js# . . . . . some . tags / / / 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": null,
  "gitCommittish": " . . . . . some . tags / / /",
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@gitlab:user..blerg--/..foo-js# . . . . . some . tags / / /",
  "rawSpec": "gitlab:user..blerg--/..foo-js# . . . . . some . tags / / /",
  "registry": undefined,
  "saveSpec": "gitlab:user..blerg--/..foo-js# . . . . . some . tags / / /",
  "scope": "@fake",
  "type": "git",
  "where": undefined,
}
`;

exports[`normalizeDepsValue gitlab:user..blerg--/..foo-js# . . . . . some . tags / / / 2`] = `"gitlab:user..blerg--/..foo-js# . . . . . some . tags / / /"`;

exports[`normalizeDepsValue gitlab:user/foo-js 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": null,
  "gitCommittish": null,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@gitlab:user/foo-js",
  "rawSpec": "gitlab:user/foo-js",
  "registry": undefined,
  "saveSpec": "gitlab:user/foo-js",
  "scope": "@fake",
  "type": "git",
  "where": undefined,
}
`;

exports[`normalizeDepsValue gitlab:user/foo-js 2`] = `"gitlab:user/foo-js"`;

exports[`normalizeDepsValue gitlab:user/foo-js#bar/baz 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": null,
  "gitCommittish": "bar/baz",
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@gitlab:user/foo-js#bar/baz",
  "rawSpec": "gitlab:user/foo-js#bar/baz",
  "registry": undefined,
  "saveSpec": "gitlab:user/foo-js#bar/baz",
  "scope": "@fake",
  "type": "git",
  "where": undefined,
}
`;

exports[`normalizeDepsValue gitlab:user/foo-js#bar/baz 2`] = `"gitlab:user/foo-js#bar/baz"`;

exports[`normalizeDepsValue gitlab:user/foo-js#bar/baz/bin 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": null,
  "gitCommittish": "bar/baz/bin",
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@gitlab:user/foo-js#bar/baz/bin",
  "rawSpec": "gitlab:user/foo-js#bar/baz/bin",
  "registry": undefined,
  "saveSpec": "gitlab:user/foo-js#bar/baz/bin",
  "scope": "@fake",
  "type": "git",
  "where": undefined,
}
`;

exports[`normalizeDepsValue gitlab:user/foo-js#bar/baz/bin 2`] = `"gitlab:user/foo-js#bar/baz/bin"`;

exports[`normalizeDepsValue https://gitlab.com/user/foo.git 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "https://gitlab.com/user/foo.git",
  "gitCommittish": null,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@https://gitlab.com/user/foo.git",
  "rawSpec": "https://gitlab.com/user/foo.git",
  "registry": undefined,
  "saveSpec": "git+https://gitlab.com/user/foo.git",
  "scope": "@fake",
  "type": "git",
  "where": undefined,
}
`;

exports[`normalizeDepsValue https://gitlab.com/user/foo.git 2`] = `"https://gitlab.com/user/foo.git"`;

exports[`normalizeDepsValue npm:@foo/bar@1.2 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": null,
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@npm:@foo/bar@1.2",
  "rawSpec": "npm:@foo/bar@1.2",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "subSpec": {
    "escapedName": "@foo%2fbar",
    "fetchSpec": "1.2",
    "gitCommittish": undefined,
    "gitRange": undefined,
    "gitSubdir": undefined,
    "name": "@foo/bar",
    "raw": "@foo/bar@1.2",
    "rawSpec": "1.2",
    "registry": true,
    "saveSpec": null,
    "scope": "@foo",
    "type": "range",
    "where": undefined,
  },
  "type": "alias",
  "where": undefined,
}
`;

exports[`normalizeDepsValue npm:@foo/bar@1.2 2`] = `"npm:@foo/bar@1.2"`;

exports[`normalizeDepsValue npm:@foo/bar@next 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": null,
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@npm:@foo/bar@next",
  "rawSpec": "npm:@foo/bar@next",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "subSpec": {
    "escapedName": "@foo%2fbar",
    "fetchSpec": "next",
    "gitCommittish": undefined,
    "gitRange": undefined,
    "gitSubdir": undefined,
    "name": "@foo/bar",
    "raw": "@foo/bar@next",
    "rawSpec": "next",
    "registry": true,
    "saveSpec": null,
    "scope": "@foo",
    "type": "tag",
    "where": undefined,
  },
  "type": "alias",
  "where": undefined,
}
`;

exports[`normalizeDepsValue npm:@foo/bar@next 2`] = `"npm:@foo/bar@next"`;

exports[`normalizeDepsValue npm:bar@1.2 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": null,
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@npm:bar@1.2",
  "rawSpec": "npm:bar@1.2",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "subSpec": {
    "escapedName": "bar",
    "fetchSpec": "1.2",
    "gitCommittish": undefined,
    "gitRange": undefined,
    "gitSubdir": undefined,
    "name": "bar",
    "raw": "bar@1.2",
    "rawSpec": "1.2",
    "registry": true,
    "saveSpec": null,
    "scope": undefined,
    "type": "range",
    "where": undefined,
  },
  "type": "alias",
  "where": undefined,
}
`;

exports[`normalizeDepsValue npm:bar@1.2 2`] = `"npm:bar@1.2"`;

exports[`normalizeDepsValue space 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "",
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@ ",
  "rawSpec": " ",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "type": "range",
  "where": undefined,
}
`;

exports[`normalizeDepsValue space 2`] = `"*"`;

exports[`normalizeDepsValue x 1`] = `
{
  "escapedName": "@fake%2ffake",
  "fetchSpec": "x",
  "gitCommittish": undefined,
  "gitRange": undefined,
  "gitSubdir": undefined,
  "name": "@fake/fake",
  "raw": "@fake/fake@x",
  "rawSpec": "x",
  "registry": true,
  "saveSpec": null,
  "scope": "@fake",
  "type": "range",
  "where": undefined,
}
`;

exports[`normalizeDepsValue x 2`] = `"x"`;
